src/main/webapp/app/exercises/programming/manage/update/update-components/custom-build-plans/programming-exercise-build-configuration/programming-exercise-build-configuration.component.ts

```
@@ -1,21 +1,80 @@
+    }
+
+    updateDockerFlags(key: string, value: string) {
+        let existingFlags = JSON.parse(this.programmingExercise()?.buildConfig?.dockerFlags || '[]') as [string, string][];
+        if (!existingFlags || existingFlags.length === 0) {
+            existingFlags = [] as [string, string][];
+        }
+        existingFlags = existingFlags.filter(([flag]) => ALLOWED_DOCKER_FLAG_OPTIONS.includes(flag) && flag !== key) || [];
+        if (ALLOWED_DOCKER_FLAG_OPTIONS.includes(key) && value.trim() !== '') {
+            existingFlags.push([key, value]);
+        }
+        this.programmingExercise()!.buildConfig!.dockerFlags = JSON.stringify(existingFlags);
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)